src/containerapp/azext_containerapp/custom.py

```diff
@@ -4332,3 +4339,258 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
         return r
     except Exception as e:
         handle_raw_exception(e)
+
+
+def patch_list(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.error("Please install or start Docker and try again.")
+        return
+    pack_exec_path = get_pack_exec_path()
+    if pack_exec_path is None:
+        return
+    print("\rListing container apps...", end="", flush=True)
+    ca_list = list_containerapp(cmd, resource_group_name, managed_env)
+    imgs = []
+    if ca_list:
+        for ca in ca_list:
+            resource_group_name = re.search('/subscriptions/[^/]+/resourceGroups/([^/]+)/', ca["id"]).group(1)
+            managed_env_name = ca["properties"]["managedEnvironmentId"].split('/')[-1]
```

Comment:
consider using environmentId

Reply:
to parse resource id, consider using parse_resource_id() to get environment name.

Reply:
This one is updated with latest commit now. But we would continue to use managed env name in our result instead of directly using environment id as a result field for user output since there's no unique identifier for managed env, and it is too long to display on terminal as well.